src/Symfony/Component/Clock/ClockAwareInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+use Psr\Clock\ClockInterface;
+
+/**
+ * An interface to help write time-sensitive classes.
+ *
+ * @author Sergii Dolgushev <p@tchwork.com>
+ */
+interface ClockAwareInterface
+{
+    public function setClock(ClockInterface $clock): void;
+
+    public function now(): DatePoint;
```

Comment:
why enforcing that a clock aware service has a **public** method exposing its `now()` to other services ? this does not make sense. The method was protected in the trait because the clock is expected to be used internally.

Forcing any clock-aware service to also be a clock themselves does not make sense.

Reply:
I see. Reverted `now` visibility change in https://github.com/symfony/symfony/pull/57710/commits/ca4da7598b67cdcc5f2c2a3243640a9015f8254f. On this note, is there any reason why it is `protected` but not `private`?